src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -904,13 +925,88 @@ private JButton getCleanTableButton() {
 				selectAll = false;
 				specificRadio.setSelected(true);
 				codeTextField.setEnabled(true);
-				inventoryRowSearchList.clear();
-				fireInventoryUpdated();
+				inventoryRowSearchList = new ArrayList<>();
+				DefaultTableModel model = (DefaultTableModel) jTableInventoryRow.getModel();
+				model.setRowCount(0);
+				model.setColumnCount(0);
+				jTableInventoryRow.updateUI();
 			}
 		});
 		return resetButton;
 	}
 
+	private JButton getValidateButton() {
+		validateButton = new JButton(MessageBundle.getMessage("angal.inventory.validate.btn"));
+		validateButton.setMnemonic(MessageBundle.getMnemonic("angal.inventory.validate.btn.key"));
+		if (inventory == null) {
+			validateButton.setEnabled(false);
+		}
+		validateButton.addActionListener(actionEvent -> {
+			if (inventory == null) {
+				MessageDialog.error(null, "angal.inventory.inventorymustsavebeforevalidation.msg");
+				return;
+			}
+			List<MedicalInventoryRow> invRowWithoutLot = inventoryRowSearchList.stream().filter(invRow -> invRow.getLot() == null).collect(Collectors.toList());
+			if (invRowWithoutLot.size() > 0) {
+				MessageDialog.error(null, "angal.inventory.allinventoryrowshouldhavelotbeforevalidation.msg");
+				return;
+			}
+			int reset = MessageDialog.yesNo(null, "angal.inventoryrow.doyoureallywanttovalidatethisinventory.msg");
+			if (reset == JOptionPane.YES_OPTION) {
+				String dischargeCode = inventory.getDischargeType();
+				String chargeCode = inventory.getChargeType();
+				Integer supplierId = inventory.getSupplier();
+				String wardCode = inventory.getDestination();
+				if (dischargeCode == null || dischargeCode.equals("")) {
+					MessageDialog.error(null, "angal.inventory.choosedischargetypebeforevalidation.msg");
+					return;
+				}
+				if (chargeCode == null || chargeCode.equals("")) {
+					MessageDialog.error(null, "angal.inventory.choosechargetypebeforevalidation.msg");
+					return;
+				}
```

Comment:
Please invert the order of these two to reflect the order in the gui
```suggestion
				if (chargeCode == null || chargeCode.equals("")) {
					MessageDialog.error(null, "angal.inventory.choosechargetypebeforevalidation.msg");
					return;
				}
				if (dischargeCode == null || dischargeCode.equals("")) {
					MessageDialog.error(null, "angal.inventory.choosedischargetypebeforevalidation.msg");
					return;
				}
```